File: Python_Ruby/stock_portfolio/portfolio.php
<?php

// Will MacLean
// CSPP 51060
// portfolio.php

$filename = "portfolio.txt";
$total = (float)0;
$shares = (float)0;
$price = (float)0;

echo "\nName \tShares \t  Price \n";
echo "---- \t------ \t  -----\n";

$file = fopen($filename, 'r') or die('goofed');
while(!feof($file))
{
	$line = fgets($file);
	//echo $line;

	list($name, $shares, $price) = split(" ", $line, 3);
	printf("%s \t %5.0f \t %6.2f \n", $name, $shares, $price);
	$total = $total + $shares * $price;

}

// report and close
echo "\nTotal Cost: ".$total." \n"; 
fclose($file);

?>

File: Python_Ruby/stock_portfolio/portfolio.rb
# Will MacLean
# CSPP 51060
# portfolio.rb

total = 0

puts "\nName \tShares \tPrice \n"
puts "---- \t------ \t----- \n"

File.readlines('portfolio.txt').each do |line|
	fields = line.split(" ", 3)
	puts fields[0] + "\t" + fields[1].to_s + "\t" + fields[2].to_s
	shares = fields[1].to_f
	price = fields[2].to_f
	total = total + shares * price
end

puts "\nTotal Cost: " + total.to_s + "\n"

